speechbrain/k2_integration/lexicon.py

```diff
@@ -0,0 +1,497 @@
+"""Lexicon class and utilities. Provides functions to read/write
+lexicon files and convert them to k2 ragged tensors. The Lexicon
+class provides a way to convert a list of words to a ragged tensor
+containing token IDs. It also stores the lexicon graph which can
+be used by a graph compiler to decode sequences.
+
+This code was adjusted from icefall's (https://github.com/k2-fsa/icefall)
+Lexicon class and its utility functions.
+
+
+Authors:
+  * Zeyu Zhao 2023
+  * Georgios Karakasidis 2023
+"""
+
+
+import logging
+import re
+import sys
+import os
+from pathlib import Path
+from typing import List, Tuple, Union
+
+try:
+    import k2
+except ImportError:
+    MSG = "Please install k2 to use k2 training \n"
+    MSG += "E.G. run: pip install k2\n"
+    raise ImportError(MSG)
+
+import torch
+
+logger = logging.getLogger(__name__)
+
+
+def get_lexicon(lang_dir, csv_files, extra_vocab_files, add_word_boundary=True):
```

Comment:
We need an example for this function so that it is tested and also easier to use by our users.